nautobot/extras/tests/test_jobs.py

```diff
@@ -1033,3 +1039,27 @@ def test_weekly_interval(self):
         schedule_day_of_week = next(iter(scheduled_job.schedule.day_of_week))
         scheduled_job_weekday = self.cron_days[schedule_day_of_week]
         self.assertEqual(scheduled_job_weekday, requested_weekday)
+
+    @override_settings(USE_TZ=True, TIME_ZONE="America/New_York")
+    def test_cron_time_zone(self):
+        self.assertEqual(app.timezone.key, "America/New_York")
+        scheduled_job = models.ScheduledJob.objects.create(
+            name="test_cron_time_zone",
+            task="pass.TestPass",
+            interval=JobExecutionType.TYPE_CUSTOM,
+            user=self.user,
+            start_time=self._strptime("1 Jan 2024 12:00AM"),
+            last_run_at=self._strptime("1 Jan 2024 9:00AM"),
+            crontab="0 9 * * *",
```

Comment:
Apologies for ignorant questions, but with `TIME_ZONE="America/New_York"`, is this crontab 0900 UTC or 0900 EST? I thought that per the original issue the user was expecting this to be evaluated as 0900 EST, but if I understand your previous reply, the `_strptime` function is evaluating the provided strings as UTC datetimes, not EST?

What part am I misunderstanding? :smile:

Reply:
Yeah this isn't fixing the problem it's just making it consistently wrong. Instead of changing `now` to `utc` we need to change `last_run_at` and `start_time` to the local time (which is how they fixed it in django celery beat)